$NetBSD: patch-ac,v 1.8 2004/07/15 04:05:40 atatat Exp $

Cast to long before int when casting a pointer.  It still "loses
precision" but in this case it doesn't matter and it quiets the
compiler.

--- process.c.orig	2003-09-18 12:53:54.017825792 +0000
+++ process.c
@@ -37,7 +37,7 @@
 #include "config.h"
 
 /* for solaris 2.1, Unixware (SVR4.2) and possibly others: */
-#ifdef SVR4
+#ifdef HAVE_SYS_STROPTS_H
 # include <sys/stropts.h>
 #endif
 
@@ -5466,7 +5466,7 @@ char *data;	/* dummy */
       *buf = 0;
       return;
     }
-  act.nr = (int)data;
+  act.nr = (int)(long)data;
   act.args = noargs;
   act.argl = 0;
   DoAction(&act, -1);
